[docs] Update ESLint guide to include info on `npx expo lint`

amandeepmittal opened the pull request:
# Why

<!--
Please describe the motivation for this PR, and link to relevant GitHub issues, forums posts, or feature requests.
-->

Add info about `npx expo lint` in ESLint guide for SDK 51 and above.

# How

<!--
How did you build this feature or fix this bug and why?
-->

By updating sections under ESLint and Prettier to include the new information and re-structure the existing information accordingly. 

# Test Plan

<!--
Please describe how you tested this change and how a reviewer could reproduce your test, especially if this PR does not include automated tests! If possible, please also provide terminal output and/or screenshots demonstrating your test/reproduction.
-->

Run docs locally and visit: http://localhost:3002/guides/using-eslint/#eslint

## Preview
![CleanShot 2024-05-06 at 03 10 17@2x](https://github.com/expo/expo/assets/10234615/2c7f3e01-52b0-42fd-8a44-09dddab3e9a1)

![CleanShot 2024-05-06 at 03 10 24@2x](https://github.com/expo/expo/assets/10234615/018b7055-516d-4441-8fb2-55b4ed5d4f1c)



# Checklist

<!--
Please check the appropriate items below if they apply to your diff. This is required for changes to Expo modules.
-->

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

amandeepmittal committed:
Update using-eslint.mdx

amandeepmittal committed:
Update using-eslint.mdx

Simek commented on docs/pages/guides/using-eslint.mdx:
```diff
@@ -3,33 +3,45 @@ title: Use ESLint and Prettier
 
 <Step label="1">
 
 Install ESLint, Prettier, and [`eslint-config-universe`](https://github.com/expo/expo/tree/main/packages/eslint-config-universe#eslint-config-universe) in your project.
```
Is there any blocker for us to recommend `eslint-config-expo` package also for older SDK versions?

In theory, it's just a ESLint config, so we should be able to use or switch it no matter of Expo SDK version, since they should be unrelated. If we go that way it should be less complicated for the users, since there will be one recommendation no matter of SDK version user uses.

amandeepmittal replied:
The `npx expo lint` command doesn't work for SDK 50 and below. I'll verify about adding `eslint-config-expo` to an SDK 50 project and let you know.

amandeepmittal replied:
Tested this and it works and can confirm there are no blockers. Let me update this guide to add eslint-config-expo specific instructions for SDK 50 and below.

Thanks for catching this!

amandeepmittal replied:
@Simek, I've updated the instructions for SDK 50 and below to install `eslint-config-expo` instead of `eslint-config-universe`. The other changes still apply for SDK 50 because `npx expo lint` to lint project doesn't work.

amandeepmittal committed:
Update using-eslint.mdx

brentvatne reviewed (commented):
looks good, my only nit is that i'd like to know what `npx expo lint` is doing for me. we could add something like:

```
# Install, configure, or run ESLint
npx expo lint

# Or install it manually
npm install ... etc
npx eslint .
```

amandeepmittal committed:
Update using-eslint.mdx

amandeepmittal commented:
@brent updated the ESLint instructions to callout in comment about when referring `npx expo lint`, what happens with the reference.

Also, added the context that running `npx expo lint` first time will also create **.eslintrc.js** (which was missing since I overlooked).

![CleanShot 2024-05-07 at 20 50 35@2x](https://github.com/expo/expo/assets/10234615/f6067749-3326-4c2c-b038-39dcfa609416)

Updated the second reference `npx expo lint` to mention what happens when one runs it second time and also included an example output of the command (from the changelog post).

![CleanShot 2024-05-07 at 20 50 42@2x](https://github.com/expo/expo/assets/10234615/f29957c9-2c34-45ed-aaff-9bc312a1e4f2)


Also, added a migration section to migrate from `eslint-config-universe` to  `eslint-config-expo` library.

![CleanShot 2024-05-07 at 21 08 33@2x](https://github.com/expo/expo/assets/10234615/53a12ffc-b3ae-40dd-b6e8-4d3c82cbc71f)

cc: @Simek

amandeepmittal committed:
Update using-eslint.mdx
